### DISCUSSION TITLE
Disable Send Globally

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#1667] [Q&A]
Our company has strict policies for file sharing for DLP. The introduction of Bitwarden Send (specifically for file sharing) has caused concern since files being shared via this platform are not inspected for DLP (Credit Card data, HIPPA, SSN, etc)

How can i GLOBALLY disable the ability to utilize this for all users?

**Comment by RealOrangeOne** [comment]
It's not possible in vaultwarden itself.

You could get around it easily by blocking `/sends` at the proxy level. It'll still show up in the UI, but none can be added. A similar thing could be added to vaultwarden to just not register the Sends endpoints when a specific envronment variable set (not especially difficult at all).

Hiding it from the UI is a bit harder.

**Reply by ghost** [reply]
Thanks. I will try the approach of blocking at the nginx layer

**Comment by dani-garcia** [comment]
For now and if you don't use attachments, you can set the maximum attachments sizes to zero, which will disable both attachments and sends.

We could add an option to disable sends in the future though.

**Reply by ghost** [reply]
Thanks for the tip. I don't want to restrict the ability to add attachments to items, so i think i will explore blocking the /sends URI at the nginx layer. It's a bit dirty, but it will fulfill the need for now.
Thanks!

**Reply by jjlin** [reply]
There is now a `sends_allowed` config setting that provides global control:

https://github.com/dani-garcia/vaultwarden/blob/8e6c6a1dc43b8993ee8814e6339d955b144d5fe9/.env.template#L59-L62

It's currently available in the `testing` images.
